### PR TITLE
Rewrite property get/put to not use ecma_op_object_get_own_property

### PR DESCRIPTION
**Pull request by zherczeg**
Continue rework the rework of ecma_op_object_get_own_property by making get/put independent from it.

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-objects.c [hunk 1]
```diff
@@ -255,11 +258,67 @@ ecma_op_object_find_own (ecma_object_t *object_p, /**< the object */
...
+                                                                 ECMA_INTERNAL_PROPERTY_ECMA_VALUE);
+
+        ecma_string_t *prim_value_str_p = ecma_get_string_from_value (*prim_value_p);
+
+        if (index < ecma_string_get_length (prim_value_str_p))
+        {
+          ecma_char_t c = ecma_string_get_char_at_pos (prim_value_str_p, index);
+          return ecma_make_string_value (ecma_new_ecma_string_from_code_unit (c));
+        }
+      }
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-objects.c]: Is it ok to fall through, if `property_name_p` is not a valid array index?

File: jerry-core/ecma/operations/ecma-objects.c [hunk 2]
```diff
@@ -255,11 +258,67 @@ ecma_op_object_find_own (ecma_object_t *object_p, /**< the object */
...
+      if (ecma_string_get_array_index (property_name_p, &index))
+      {
+        ecma_value_t *prim_value_p = ecma_get_internal_property (object_p,
+                                                                 ECMA_INTERNAL_PROPERTY_ECMA_VALUE);
+
+        ecma_string_t *prim_value_str_p = ecma_get_string_from_value (*prim_value_p);
+
+        if (index < ecma_string_get_length (prim_value_str_p))
+        {
+          ecma_char_t c = ecma_string_get_char_at_pos (prim_value_str_p, index);
```
Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-objects.c]: `c` -> `char_at_idx` ?

File: jerry-core/ecma/operations/ecma-objects.c (hunk 1 above)
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-objects.c]: Yes, those property names are just regular properties. E.g: 

    var a = new String("x")
    a[0] = 1; // nothing happens
    a[1] = 1; // a[1] is defined with a value of 1

Comment (LaszloLango) [inline jerry-core/ecma/operations/ecma-objects.c]: I see, thanks for the explanation.

**Comment by LaszloLango** [comment]
LGTM, after the variable name is fixed.

**Comment by dbatyai** [comment]
LGTM

**Comment by zherczeg** [comment]
Thank you.

**Pull request merged**